beacon_node/lighthouse_network/src/peer_manager/peerdb/peer_info.rs

```diff
@@ -504,6 +501,50 @@ impl<E: EthSpec> PeerInfo<E> {
         }
     }
 
+    fn compute_custody_subnets_from_metadata(
+        &mut self,
+        node_id: NodeId,
+        peer_id: &PeerId,
+        metadata: &MetaData<E>,
+        spec: &ChainSpec,
+        log: &Logger,
+    ) -> HashSet<DataColumnSubnetId> {
+        let node_id = node_id.raw().into();
+
+        // fallback to `custody_requirement` if `csc` is invalid
+        let custody_subnet_count = match metadata.custody_subnet_count() {
+            Ok(csc) => *csc,
+            Err(_) => {
```

Comment:
This Err condition will only occur if MetaData is not a V3 variant. Invalid values will trigger an error on the ReqResp SSZ decoding step. Can we ensure that when calling this function for a peer we have already issued a metadata/v3 request? I think here we can default to `custody_requirement` without any log as it's our fault to not issue a `metadata/v3`. Or maybe signal this fact? i.e. `Expected MetaDataV3 for peer`

Reply:
Good catch. I've added a check for `is_peer_das_scheduled` before calling this function, and add `Expected MetaDataV3` to the log. 

I think it should be an error log if it's a bug though.

Reply:
Updated in https://github.com/sigp/lighthouse/pull/6308/commits/e8b9d769ac92d03565f8b1ec03fbfd68c2f987f8